File: problems/java-reverse-string/src/test/java/StringReverserTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class StringReverserTest {
    @Test
    public void testBasicString() {
        assertEquals("olleh", StringReverser.reverse("hello"));
    }

    @Test
    public void testEmptyString() {
        assertEquals("", StringReverser.reverse(""));
    }

    @Test
    public void testNullInput() {
        assertEquals("", StringReverser.reverse(null));
    }

    @Test
    public void testSingleCharacter() {
        assertEquals("a", StringReverser.reverse("a"));
    }

    @Test
    public void testUnicode() {
        assertEquals("界世好你", StringReverser.reverse("你好世界"));
    }

    @Test
    public void testPalindrome() {
        assertEquals("abcba", StringReverser.reverse("abcba"));
    }
}

File: problems/dotnet-reverse-string/StringReverserTest.cs
using System;
using Xunit;

public class StringReverserTest
{
    [Fact]
    public void TestBasicReverse()
    {
        Assert.Equal("olleh", StringReverser.Reverse("hello"));
    }

    [Fact]
    public void TestEmptyString()
    {
        Assert.Equal("", StringReverser.Reverse(""));
    }

    [Fact]
    public void TestNullString()
    {
        Assert.Equal("", StringReverser.Reverse(null));
    }

    [Fact]
    public void TestSingleCharacter()
    {
        Assert.Equal("a", StringReverser.Reverse("a"));
    }

    [Fact]
    public void TestWithSpaces()
    {
        Assert.Equal("dlrow olleh", StringReverser.Reverse("hello world"));
    }

    [Fact]
    public void TestPalindrome()
    {
        Assert.Equal("racecar", StringReverser.Reverse("racecar"));
    }

    [Fact]
    public void TestWithNumbers()
    {
        Assert.Equal("321", StringReverser.Reverse("123"));
    }

    [Fact]
    public void TestMixedContent()
    {
        Assert.Equal("!dlroW olleH", StringReverser.Reverse("Hello World!"));
    }
}
